File: SiteVisitsNetCore2020Web/ClientApp/src/app/models/PageUrlVariation.ts
import { PageUrl } from './PageUrl';

export class PageUrlVariation {
    urlVariation: string;
    urlVariationCompact: string;
    pageUrl: PageUrl;
  constructor(options: {
    urlVariation: string,
    urlVariationCompact: string,
    pageUrl: PageUrl
  }) {
    this.urlVariation = options.urlVariation;
    this.urlVariationCompact = options.urlVariationCompact;
    this.pageUrl = new PageUrl(options.pageUrl);
  }

    }

File: SiteVisitsNetCore2020Web/Models/PageUrlVariation.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;

namespace SiteVisitsNetCore2020Web.Models
{
    public class PageUrlVariation
    {
        [Key]
        public string UrlVariation { get; set; }
        public string UrlVariationCompact { get; set; }
        public DateTime FirstOccurrence { get; set; }
        public virtual PageUrl PageUrl { get; set; }
        /**
        protected IAdapter adapter;
        public PageUrlVariation() { }

        public PageUrlVariation(IAdapter adapterV)
        {
            adapter = adapterV;
        }

        public void AddToContext()
        {
            adapter.AddPageUrlVariation(this);
        }
**/
    }
}
